### PR TITLE
Add a texture mipmap case

### PR DESCRIPTION
**Pull request by jchen10**
To catch a bug on MacOSX. glTexStorage* don't work as expected if
the current GL_TEXTURE_BASE_LEVEL is not 0. As a result, the shader
function textureSize returns wrong size.

### CODE REVIEW

File: sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html [hunk 1]
```diff
@@ -59,6 +59,28 @@
     gl_FragColor = texture2D(tex, texCoord);
 }
 </script>
+
+<script id="vshader_texsize" type="x-shader/x-vertex">#version 300 es
+uniform vec4 uMult;
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html]: What's the point of uMult here? Looks irrelevant to me.

File: sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html [hunk 2]
```diff
@@ -189,6 +211,30 @@
...
+
+  // Test textureSize should work correctly with non-zero base level
+  var program = wtu.setupProgram(
+      gl, ['vshader_texsize', 'fshader_texsize'], ['vPosition'], [0]);
+
+  gl.uniform1i(gl.getUniformLocation(program, "tex"), 0);
+  gl.uniform4f(gl.getUniformLocation(program, "uMult"), 1, 1, 1, 1);
+  gl.uniform1i(gl.getUniformLocation(program, "lod"), 1);
+  gl.uniform2i(gl.getUniformLocation(program, "texSize"), 7, 4);
+  tex = gl.createTexture();
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html]: My understanding is lod0 is (31, 17), lod1 is (15, 8).

File: sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html [hunk 3]
```diff
@@ -189,6 +211,30 @@
...
+  gl.bindTexture(gl.TEXTURE_2D, tex);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.LINEAR);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texParameter(TEXTURE_MAG_FILTER) should succeed");
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texParameter(TEXTURE_MIN_FILTER) should succeed");
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_BASE_LEVEL, 1);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texParameter(TEXTURE_BASE_LEVEL) should succeed");
+  gl.texStorage2D(gl.TEXTURE_2D, 4, gl.RGBA8, 31, 17);
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html]: Can you also test texStorage3D?

File: sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html (hunk 2 above)
Comment (kenrussell) [inline sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html]: Right, but the base level is being set to 1 below, so the call to textureSize for level 1 actually fetches lod 1+1 = 2, which would be (7, 4) -- if I'm understanding the test correctly.

File: sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html (hunk 3 above)
Comment (kenrussell) [inline sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html]: Is there a reason to use these odd numbers? Why not use a size that at least is evenly divisible by 4 in both dimensions?

**Comment by kenrussell** [comment]
Thanks for this test. On the whole it looks good, but please address @zhenyao 's comments and let him merge it.

File: sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html (hunk 2 above)
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html]: You are right.

File: sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html (hunk 1 above)
Comment (jchen10) [inline sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html]: Yes, it can be omitted. It was just because of the existing code around. Fixed anyway!

File: sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html (hunk 3 above)
Comment (jchen10) [inline sdk/tests/conformance2/textures/misc/tex-mipmap-levels.html]: Covered in the newly added texStorage3D case!

**Comment by zhenyao** [comment]
lgtm
but @kenrussell should take another look

**Comment by kenrussell** [comment]
LGTM too. Thanks for the expanded tests.
